src/FSharp.Stats/Fitting/GeneralisedLinearModel.fs

```diff
@@ -0,0 +1,697 @@
+/// </description>
+/// </item>
+/// </list>
+/// </remarks>
+type GLMStatisticsPrameter = 
+    {
+        //Name:string
+        /// The coefficient of the parameter.
+        Coefficient: float
+        /// The standard error of the parameter.
+        StandardError: float
+        /// The Z-score of the parameter.
+        ZScore: float
+        /// The person of Z of the parameter.
+        PersonOfZ: float
+    }
+
+/// This module contains various link functions used in generalized linear models.
+module LinkFunctions =
+    /// Clips the logistic values to avoid numerical instability.
+    let internal clipLogisticValues (p : float)  = 
+        let floatEps = 2.220446049250313e-16
+
+        max floatEps (min (1.0-floatEps) p)
+
+    /// Clips the logistic values to avoid numerical instability.
+    let internal clipLogisticValues2 (p : float)  = 
+        let floatEps = 2.220446049250313e-16
+
+        max floatEps p
+
+    /// The logit link function used in logistic regression.
+    let LogitLinkFunction : LinkFunction =
+        {
+            // Computes the link function value for a given parameter.
+            getLink = fun b -> 
+                let p = clipLogisticValues b
+                System.Math.Log(p / (1.0 - p))
+            // Computes the inverse link function value for a given parameter.
+            getInvLink = fun a -> 
+                1.0 / (1.0 + System.Math.Exp(-a))
+            // Computes the derivative of the link function for a given parameter.
+            getDeriv = fun a ->
+                let p = clipLogisticValues a
+                1./(p*(1.-p))
+            // Computes the derivative of the inverse link function for a given parameter.
+            getInvLinkDerivative = fun a ->
+                let t = System.Math.Exp(a)
+                t / ((1.0 + t) * (1.0 + t))
+        }
+
+    /// The log link function used in Poisson regression.
+    let LogLinkFunction : LinkFunction =
+        {
+            // Computes the link function value for a given parameter.
+            getLink = fun b -> System.Math.Log((clipLogisticValues2 b))
+            // Computes the inverse link function value for a given parameter.
+            getInvLink = fun a -> System.Math.Exp(a)
+            // Computes the derivative of the link function for a given parameter.
+            getDeriv = fun a -> 1./(clipLogisticValues2 a) 
+            // Computes the derivative of the inverse link function for a given parameter.
+            getInvLinkDerivative = fun a -> System.Math.Exp(a)
+        }
+
+    /// The inverse squared link function used in gamma regression.
+    let InverseSquaredLinkFunction: LinkFunction =
+        {
+            // Computes the link function value for a given parameter.
+            getLink = fun b -> Math.Pow(b,-2.)//1.0 / b
+            // Computes the inverse link function value for a given parameter.
+            getInvLink = fun a -> Math.Pow(a,(1./ -2.))//1.0 / a
```

Comment:
I assume the `1./2.` and later `-1.-1.` are left on purpose to make the implementation easier to understand :+1: